### ISSUE TITLE
AppNavMenu: Changes Header After 'CanDeactivate' Triggered During Routing

### ISSUE THREAD

**Issue by anton-kurilov** [infor-design/enterprise-ng#1553]
**Describe the bug**
Issue is easy to reproduce here: https://stackblitz.com/edit/ids-quick-start-1651-yfireh?file=src%2Fapp%2Fservices%2Fcandeactivate.service.ts 

observer that fhir-options, and fhir-resources pages navigate to both the correct header, component and url
switch to fhir-options, put in some text in the textfield, and click cancel to stay on the same component, observe that the header changed anyways.

**Expected behavior**
Header should not change before CanDeactivate has been executed 

**Version**
per stackbiltz

**Screenshots**
![image](https://github.com/infor-design/enterprise-ng/assets/14839675/8a9f0d78-845a-4537-b6c4-18598200219f)

**Platform**
See stackblitz

**Additional context**
Possible workaround to not use an alert or warning to the user. Yes sure it is possible, but our components should behave nicer with built in angular routing, including guards they expose.
CanDeactiviate is deprecated, maybe there is a better way to do this now?
See if it can be reproduced on the newer module-nav

**Comment by tmcconechy** [comment]
Being that this is deprecated could try https://angular.io/guide/router-tutorial-toh#canactivate-requiring-authentication
I see this as an angular "question" more than a bug in the app nav.

What about "navigating back" if you hit cancel?

**Comment by anton-kurilov** [comment]
@tmcconechy we tried navigating back, but doesn't really work, because you never actually navigate away, so same issue is present. Then there were some hacky attempts to refire the click event to get back to the current menu selection and stay on the same route, but that seemed to open up a can of worms (weird 'active' state styling on menu)

**Comment by tmcconechy** [comment]
Yeah, It has to NOT navigate i guess. Unfortunately have not used the angular router enough to help well on this. Might also be worth a question in stack overflow directed to angular router people . As i think it is angular logic... But not 100% sure

**Comment by tmcconechy** [comment]
@anton-kurilov didnt you find a solution for this? as per https://teams.microsoft.com/l/message/19:83173693057f4991b7a84f030975c06d@thread.skype/1694443823157?tenantId=457d5685-0467-4d05-b23b-8f817adda47c&groupId=4f50ef7d-e88d-4ccb-98ca-65f26e57fe35&parentMessageId=1694443823157&teamName=IDS%20Enterprise%20Development&channelName=Angular%20Components&createdTime=1694443823157

So can we close

**Comment by anton-kurilov** [comment]
@tmcconechy, not quite, it solved like 1/2 the issue, even with the workaround we implemented still note that the menu item selection still changes even though the route didn't.

Even so it might be worth it to keep this open and see if we can create a demo page for this? Since it impacts a base component with built in angular routing controls, no real business logic.
